### PR TITLE
Add chunk_model as parameter

### PR DESCRIPTION
**Pull request by LouisLeNezet**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/phaseimpute/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/phaseimpute _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by LouisLeNezet** [comment]
I've retried on my cluster and the fulltest does now work without any issue with the modification on GLIMPSE2_chunk.

File: nextflow.config [hunk 1]
```diff
@@ -24,6 +24,7 @@ params {
     normalize                   = true
     compute_freq                = false
     remove_samples              = null
+    chunk_model                 = 'sequential'
```
Comment (atrigila) [inline nextflow.config:27]: There is no test assessing that the sequential mode works.

File: nextflow_schema.json [hunk 2]
```diff
@@ -132,6 +132,13 @@
                 "binaryref": {
                     "type": "string",
                     "description": "Whether to generate a binary reference file to be used with GLIMPSE2"
+                },
+                "chunk_model": {
+                    "type": "string",
+                    "description": "Model type to use for GLIMPSE2_CHUNK",
+                    "enum": ["recursive", "sequential", "uniform-number-variants"],
```
Comment (atrigila) [inline nextflow_schema.json:139]: There should be a test assessing each of these modes (at least the new ones, sequential and uniform-number-variants).

File: subworkflows/local/vcf_chunk_glimpse/tests/main.nf.test
Comment (atrigila) [inline subworkflows/local/vcf_chunk_glimpse/tests/main.nf.test:1]: Test the `sequential` and `uniform-number-variants modes`.

File: conf/steps/panel_prep.config [hunk 3]
```diff
@@ -184,6 +185,7 @@ process {
     }
 
     withName: 'NFCORE_PHASEIMPUTE:PHASEIMPUTE:VCF_CHUNK_GLIMPSE:GLIMPSE_CHUNK' {
+        ext.args   = "--window-size 4"
```
Comment (atrigila) [inline conf/steps/panel_prep.config:188]: Adding this new argument does not change any results in the tests?

File: workflows/phaseimpute/main.nf [hunk 4]
```diff
@@ -171,14 +171,15 @@ workflow PHASEIMPUTE {
...
+                ch_map,
+                params.chunk_model
             )
             ch_panel_phased = VCF_PHASE_SHAPEIT5.out.vcf_tbi
             ch_versions = ch_versions.mix(VCF_PHASE_SHAPEIT5.out.versions)
         }
 
         // Create chunks from reference VCF
-        VCF_CHUNK_GLIMPSE(ch_panel_phased, ch_map)
+        VCF_CHUNK_GLIMPSE(ch_panel_phased, ch_map, params.chunk_model)
```
Comment (atrigila) [inline workflows/phaseimpute/main.nf:182]: There should be no `params.` in the `main.nf`. They should be in `pipeline_initialisation`.

**Review by atrigila — CHANGES_REQUESTED** [review]
The idea is good, some tests are missing in order to use sequential as the default chunk model for the pipeline. We should assert that it really works, otherwise, we will have the same problem when testing with AWS mega tests.

File: workflows/phaseimpute/main.nf (hunk 4 above)
Comment (LouisLeNezet) [reply workflows/phaseimpute/main.nf:182]: This should be done now

File: conf/steps/panel_prep.config (hunk 3 above)
Comment (atrigila) [reply conf/steps/panel_prep.config:188]: Makes sense!

File: nextflow_schema.json (hunk 2 above)
Comment (LouisLeNezet) [reply nextflow_schema.json:139]: I will create them in the subworkflows.

File: nextflow.config (hunk 1 above)
Comment (LouisLeNezet) [reply nextflow.config:27]: For small regions it seems that the 'sequential' mode create more than two overlapping regions at the start of the chromosome, however it is the recommended model for large region.

File: nextflow_schema.json (hunk 2 above)
Comment (atrigila) [reply nextflow_schema.json:139]: Great!

File: nextflow.config (hunk 1 above)
Comment (atrigila) [reply nextflow.config:27]: Fair enough! and we test this with the full sized test right? So there is no need to add a small test here as it is at least covered by the full size test.

Comment (LouisLeNezet) [reply nextflow.config:27]: I agree !
I've tried it on my HPC and it worked well with this parameter.

File: subworkflows/local/vcf_chunk_glimpse/tests/main.nf.test
Comment (LouisLeNezet) [reply subworkflows/local/vcf_chunk_glimpse/tests/main.nf.test:1]: Apparently `uniform-number-variants` is not yet available.
I will remove it

**Review by atrigila — APPROVED** [review]

**Pull request merged**